dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -17,95 +17,167 @@
 
 package com.intel.analytics.bigdl.nn
 
-import com.intel.analytics.bigdl.Module
+import com.intel.analytics.bigdl.nn.abstractnn.Activity
 import com.intel.analytics.bigdl.tensor.Tensor
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
-import com.intel.analytics.bigdl.utils.T
+import com.intel.analytics.bigdl.utils.{T, Table}
 
 import scala.reflect.ClassTag
 
 class Recurrent[T : ClassTag] (
-  hiddenSize: Int = 3,
-  bpttTruncate: Int = 2)
+  hiddenSize: Int = 3)
   (implicit ev: TensorNumeric[T]) extends Container[Tensor[T], Tensor[T], T] {
 
-  val hidden = Tensor[T]()
-  var module: Module[T] = _
-  var transform: Module[T] = _
+  var hidden: Activity = null
+  val batchDim = 1
+  val timeDim = 2
+  val hidDim = 1
+  val inputDim = 2
   var (batchSize, times) = (0, 0)
 
+  override def reset(): Unit = {
+    modules(0).reset()
+    modules(0).isInstanceOf[RnnCell[T]] match {
+      case true => hidden = Tensor[T]()
+      case _ => throw new IllegalArgumentException("Cell not implemented")
```

Comment:
Maybe we should have the cell to return a zero-valued hidden to initialize the value?